### DISCUSSION TITLE
Why does poetry force the user to set package attributes?

### DISCUSSION THREAD

**Discussion by EvanShaw** [python-poetry/poetry#7543] [General]
I understand that a README file, the authors field, version, etc. are required for _publishing a package._

But, what if I only want to use poetry for dependency management? It seems like a completely valid way to use poetry, but the enforcement of these fields makes it seem like it isn't. Other package managers such as NPM, Composer, etc. don't make the strict assumption that the user is publishing a package.

I'm currently working in a monorepo with several different AWS CDK stacks which all manage their own dependencies with poetry, but it seems pretty silly that each and every one of these must have a README, authors, etc...

Any reason why poetry is designed this way?

**Comment by PabloEmidio** [comment]
I understand that attributes as author and version are not only for published packages, but a general attribution to a project. In given moment, a project had its version, maintainers, should know info (readme).

And, speaking strictly about poetry, it define itself as ` Python packaging and dependency management made easy `. Note the `and` in the setence; it is not a `one or other`, but `one and other.

Also, it's interesting to know that `poetry` is one of the tools which came to implement the new setup notation for a python project.
https://peps.python.org/pep-0517/
https://peps.python.org/pep-0518/
 
And, in the future versions.
https://peps.python.org/pep-0621/
